swabbie-core/src/main/kotlin/com/netflix/spinnaker/swabbie/events/ResourceStateManager.kt

```diff
@@ -162,37 +163,68 @@ class ResourceStateManager(
     if (event is OptOutResourceEvent) {
       log.debug("Tagging resource ${event.markedResource.uniqueId()} with \"expiration_time\":\"never\"")
       val taskId = tagResource(event.markedResource, event.workConfiguration)
-      log.debug("Tagging resource ${event.markedResource.uniqueId()} in {}", StructuredArguments.kv("taskId", taskId))
+      if (taskId != null) {
+        log.debug("Tagging resource ${event.markedResource.uniqueId()} in {}", StructuredArguments.kv("taskId", taskId))
+      }
     }
   }
 
   // todo eb: pull to another kind of ResourceTagger?
   private fun tagResource(
     resource: MarkedResource,
     workConfiguration: WorkConfiguration
-  ): String {
-    val taskId = taggingService.upsertImageTag(
-      UpsertImageTagsRequest(
-        imageNames = setOf(resource.name ?: resource.resourceId),
+  ): String? {
+    var taskId: String? = with(resource.resourceType) {
+      when {
+        equals("serverGroup", true) -> tagAsg(resource, workConfiguration)
+        equals("image", true) -> tagImage(resource)
+        else -> {
+          log.error("Failed to tag resource ${resource.uniqueId()} with aws expiration_time tag ")
```

Comment:
This is not the right message - it's that we have no _way_ to tag the resource because we don't know how, or it's not supported. I'd change to something like:

```
log.warn("Resource type $it cannot be tagged with an infrastructure tag.") 
```
Or something like that. (not sure if the `$it` syntax is right when accessing that resource type).

Reply:
updated with a better log message